pkg/storage/plugins/mongodb_docker/store.go

```diff
@@ -154,15 +230,15 @@ func EnsureMongoIsRunning(c *portercontext.Context, container string, port strin
 	for {
 		select {
 		case <-timeout.Done():
-			return nil, errors.New("timeout waiting for local mongodb daemon to be ready")
+			return nil, span.Error(errors.New("timeout waiting for local mongodb daemon to be ready"))
 		default:
 			conn := mongodb.NewStore(c, mongoPluginCfg)
 			err := conn.Connect(ctx)
 			if err == nil {
 				return conn, nil
-			} else if c.Debug {
-				fmt.Fprintln(c.Err, errors.Wrapf(err, "mongodb isn't ready yet"))
 			}
+
+			time.Sleep(time.Second)
```

Comment:
Why do we need a Sleep here?

Reply:
I thought that it probably wasn't great to have a tight loop without any yields constantly checking the progress of mongodb starting up, which takes cycles away from ... mongo starting up. Perhaps a second isn't the right amount of time to sleep, but I do think we should sleep here for some amount of time.

Reply:
Maybe using `time.Ticker` is better suited for implementing a backoff here? I know this is minor but using a time.Sleep here may add delay for the 10 second time out we set for trying to connect to the db

Reply:
I'm not sure I'm following why ticker would work fundamentally different than the loop + sleep here?

It's only sleeping 50 milliseconds at each iteration of the loop (to give mongo time to start up). During sleep, the goroutine yields so it's not keeping the CPU busy. Is there something I'm missing?

Reply:
Yes, if I understand correctly, if the time out is about to happen right before we hit the sleep call. Then we would be waiting for another 50 milliseconds. If we used ticker in the `select` statement, the select would immediately return since the done channel would be closed. The code would look like this
```golang
d := time.NewTicker(50 * time.millisecond)
defer d.Stop()
for {
     select {
	case <-timeout.Done():
		return nil, span.Error(errors.New("timeout waiting for local mongodb daemon to be ready"))
         case <-d.C:
		conn := mongodb.NewStore(c, mongoPluginCfg)
		err := conn.Connect(ctx)
		if err == nil {
			return conn, nil
		}
      }
 }
```
The performance here probably isn't critical. I would be fine to merge as it is

Reply:
Oh! Thanks for explaining. Yeah I can switch to a ticker so that we absolutely stop waiting when we hit our deadline.